src/functions_framework/event_conversion.py

```diff
@@ -168,3 +179,50 @@ def _split_resource(context: Context) -> Tuple[str, str, str]:
         raise EventConversionException("Resource regex did not match")
 
     return service, match.group(1), match.group(2)
+
+
+def marshal_background_event_data(request):
+    """Marshal the request body of a raw Pub/Sub HTTP request into the schema that is expected of
+    a background event"""
+    request_data = request.get_json()
+    if not _is_raw_pubsub_payload(request_data):
+        # If this in not a raw Pub/Sub request, return the unaltered request data.
+        return request_data
+    
+    return {
+        "context": {
+            "eventId": request_data["message"]["messageId"],
+            "timestamp": datetime.utcnow().isoformat() + "Z",
+            "eventType": _PUBSUB_EVENT_TYPE,
+            "resource": {
+                "service": _PUBSUB_CE_SERVICE,
+                "type": _PUBSUB_MESSAGE_TYPE,
+                "name": _parse_pubsub_topic(request.path),
+            },
+        },
+        "data": {
+            '@type': _PUBSUB_MESSAGE_TYPE,
+            "data": request_data["message"]["data"],
+            "attributes": request_data["message"]["attributes"],
+        }
+    }
+
+
+def _is_raw_pubsub_payload(request_data) -> bool:
+    """Does the given request body match the schema of a unmarshalled Pub/Sub request"""
+    return (
+        request_data is not None and
+        "context" not in request_data and
+        "subscription" in request_data and
+        "message" in request_data and
+        "data" in request_data["message"] and
+        "messageId" in request_data["message"]
+    )
+
+
+def _parse_pubsub_topic(request_path) -> Optional[str]:
+    match = _PUBSUB_TOPIC_REQUEST_PATH.search(request_path)
+    if match:
+        return match.group(0)
+    else:
+        return None
```

Comment:
Is this actually something that would happen in the wild, and have meaning? Or just happen with the emulator? Should we just return an empty string here (and remove the `None`/empty-string conversion above) instead?

Reply:
It could happen in the wild if a customer manually creates a Pub/Sub subscription and points `pushEndpoint` at their function's URL. I can update to an empty string. 

The other option is to fall back to using the subscription name, as that will always be included in the Pub/Sub payload. I wavered on whether or not that was a good idea...

Reply:
This one: go/link-to-the-test-for-matt

Reply:
I think it makes sense to return an empty string here and revert the conversion above.